src/aunit/Test.cpp

```diff
@@ -76,25 +76,20 @@ void Test::resolve() {
   if (!isVerbosity(Verbosity::kTestAll)) return;
 
   Print* printer = Printer::getPrinter();
+  printer->print(TEST_STRING);
```

Comment:
Good attempt, but now the behavior is subtly different when `mStatus == Test::kStatusUnknown`, right? The current code prints nothing, but your code now prints "Test ", without a trailing newline, which I think is a bit worse. The code duplication should be inconsequential on a desktop, so the memory savings is relevant only to an 8-bit processor, and I think even there, I think the flash memory savings will amount to about 3 x (10-15) bytes, or about 30-50 bytes.

Reply:
Yes ... and this is exactly why you should use enum instead of const ints. The compiler sends a warning when unused enums are not in any case.
This is exactly what I'm doing for AUnitPC (I'm gonna rename it with this name I guess).

Reply:
By the way, doing the changes, I've found that kStatusDone in documentation does not exists. Same for kStatusSetup if I'm not wrong.

Reply:
enum class is probably safer, but I created AUnit before I learned about it, and I didn't have motivation to go back to this, since most (if not all) of this is internal and not exposed to the end-user except through the Failed/Passed status messages. The `LifeCycle` and `Status` variables actually started as a single variable, and only later did it occur to me that it makes more sense to split them, so there might be some residual commingling between them.

Funny thing about `enum class`, when I learned about it, I started using it, then I started to find them a bit annoying and unergonomic, so I found myself slowly shifting back to using just `static constexpr` for internal things. I don't know, I've started to write more C and Golang, and neither of them have enums, so I guess they caused a shift in my thinking process. I think it's all part of my "C++ has too much syntax and cognitive overhead" annoyance.

Reply:
Correction: C has `enum`, not `enum class`, but if I recall, C-enum is in the global namespace and doesn't introduce a new type, so basically the same as an `int`